Packet++/src/SipLayer.cpp

```diff
@@ -208,6 +296,85 @@ namespace pcpp
 		return methodAdEnum->second;
 	}
 
+	SipRequestFirstLine::SipFirstLineData SipRequestFirstLine::parseFirstLine(const char* data, size_t dataLen)
+	{
+		SipFirstLineData result = { "", "", "", false };
+
+		if (data == nullptr || dataLen == 0)
+		{
+			PCPP_LOG_DEBUG("Empty data in SIP request line");
+			return result;
+		}
+
+		// Find first space (end of METHOD)
+		size_t firstSpaceIndex = 0;
+		while (firstSpaceIndex < dataLen && data[firstSpaceIndex] != ' ')
+		{
+			firstSpaceIndex++;
+		}
+
+		if (firstSpaceIndex == 0 || firstSpaceIndex == dataLen)
+		{
+			PCPP_LOG_DEBUG("Invalid METHOD in SIP request line");
+			return result;
+		}
+
+		// Validate method exists in SipMethodStringToEnum
+		std::string methodStr{ data, firstSpaceIndex };
+		auto methodIt = SipMethodStringToEnum.find(methodStr);
+		if (methodIt == SipMethodStringToEnum.end())
+		{
+			PCPP_LOG_DEBUG("Unknown SIP method");
+			return result;
+		}
+
+		// Find second space (end of URI)
+		size_t secondSpaceIndex = firstSpaceIndex + 1;
+		while (secondSpaceIndex < dataLen && data[secondSpaceIndex] != ' ')
+			secondSpaceIndex++;
+
+		if (secondSpaceIndex == dataLen)
+		{
+			PCPP_LOG_DEBUG("No space before version");
+			return result;
+		}
+
+		size_t uriLen = secondSpaceIndex - firstSpaceIndex - 1;
+		if (uriLen == 0)
+		{
+			PCPP_LOG_DEBUG("Empty URI");
+			return result;
+		}
+
+		// Find end of line
+		size_t lineEnd = secondSpaceIndex + 1;
+		while (lineEnd < dataLen && data[lineEnd] != '\r' && data[lineEnd] != '\n')
+			lineEnd++;
+
+		// Minimum length for "SIP/x.y"
+		size_t versionLen = lineEnd - secondSpaceIndex - 1;
+		if (versionLen < 7)
+		{
+			PCPP_LOG_DEBUG("Version too short");
+			return result;
+		}
+
+		const char* versionStart = data + secondSpaceIndex + 1;
+		if (versionStart[0] != 'S' || versionStart[1] != 'I' || versionStart[2] != 'P' || versionStart[3] != '/')
+		{
+			PCPP_LOG_DEBUG("Invalid SIP version format");
+			return result;
+		}
+
+		// All validations passed
+		result.method = methodStr;
```

Comment:
nit: Use move assignment to avoid a copy.
```suggestion
		result.method = std::move(methodStr);
```

Reply:
Thanks for the correction and assembly proof! You're right - `std::move` is needed here. The assembly clearly shows copy vs. move assignment. My mistake!